src/command/resume_only.rs

```diff
@@ -123,10 +127,6 @@ fn handle_finished_swap(
 
     let swap_id = finished_swap.swap.swap_id();
 
-    let _ = db
-        .remove_active_taker(&finished_swap.taker)
```

Comment:
Can you explain please?

Reply:
I removed the code which saves active takers to the db so this wont do anything. The new order matching engine has no concept of Takers. I am not 100% how this affects nectar.

Reply:
Ok, allow me to give you a bit more context.

To mitigate DDOS type attack, Nectar does not execute 2 swaps with the same counterpart.
Hence, we store `takers` in the DB when starting a swap, and delete them when the swap is finished.

When a new swap is meant to start, we check if a swap is already ongoing with the given counterparty. If it is, then we do not start the swap.

I believe that this still fits in the new design. If 2 parties match orders, then it's ok to have a consensus rule to not start a new swap if they are already executing one.

I believe this feature is good to have and if we were to remove it, it should be by choice.

I'd be keen to keep the feature. @thomaseizinger @bonomat please share your thoughts,

Reply:
I agree, it is still a good feature to keep. 
It does not give you a 100% guaranty that you don't get gamed but it makes it at least harder.

Reply:
Yep, it looks like we just might need to change some terminology at some point!

Reply:
Yes, feel free to change to `Counterpart` here or just add a TODO to rename later. Either way, I think the feature needs to be kept and I understand it follow the new matching consensus principle.

Reply:
Added this functionality back in and renamed it

Reply:
@rishflab: I see your renamed `Taker` to `Peer` but I think you missed the re-addition of the call here to remove from the db once a swap is finished here.